README.md

```diff
@@ -10,57 +9,86 @@ What Currently Works
+Protip: Use a VM (VMWare, VirtualBox). If you mess up, simply hose it and start squeaky clean. Or you can use an ARM based linux box (R-Pi, ODROID, etc).
 
-* PHP is [an ungodly clusterfuck](http://eev.ee/blog/2012/04/09/php-a-fractal-of-bad-design/) that's insanely unwieldly, performs about as well as a football team in Antarctica and is a nightmare to maintain.  
-  I've written large applications in PHP before, and I'm not going there again.
-
-* Ruby and Java are both massive resource hogs, and this server needs to be able to scale well, preferably without me having me hire another server I can't pay for.  
-  I run a JIRA (Java) and a Discourse (Ruby) instance, and they both chew up nearly everything their servers have to give.
+### Q: Why is it built on Python?
+There aren't actually that many things you can build a web application on, and most of them were out from the start.  
+There was a large rant by one the kcsrv owner, I (SoftwareGuy) who improved this FAQ have condensed it down to make it less ranty.
 
+* PHP is [an ungodly clusterfuck](http://eev.ee/blog/2012/04/09/php-a-fractal-of-bad-design/).
+* Ruby and Java are both massive resource hogs, and this server needs to be able to scale well. Ruby / Java seem to eat the server alive.
 * NodeJS is actually a good candidate, mostly thanks to being blazing fast.  
-  Unfortunately, nearly everything around it (including the runtime itself) is in beta, and many things (such as database abstraction layers) simply aren't mature enough for me to want to depend on it - I have yet to see a robust migration system, for instance.  
-  One day I might even port over at least parts of the server to it (the game API comes to mind), but not right now.
+  Unfortunately, nearly everything around it (including the runtime itself) is in beta.
 
-Python isn't the fastest thing around, but both it and its ecosystem are very stable and mature. SQLAlchemy + Alembic alone makes an incredibly robust database abstraction layer, that nothing I've seen in NodeJS can match. (Do feel free to prove me wrong though.)
+In short, Python isn't the fastest thing around, but both it and its ecosystem are very stable and mature. SQLAlchemy + Alembic alone makes an incredibly robust database abstraction layer.
 
 
 Development Setup
 ---
-
-Probably only works on Linux. Screw Ansible/Vagrant, too hard for me.
+Make sure you have postgresql, postgresql development libs, python3, python3-pip python3-requests, libcurl-openssl-dev and virtualenv packages installed. Consult your package manager for those packages to be installed (yum, apt-get...)
 
 1.  **Set up the environment.**  
-
-        pip install -r requirements.txt
-        sudo -u postgre createuser -s $USER
-        sudo -u postgre createdb kcsrv
-
+Only run virtualenv once. You'll break the environment if you do it every time! (If you're booting the server after first time, just skip this line)
```

Comment:
Suggested wording for this section:

---

1. **Set up the environment**
    
        git clone http://github.com/kancolletool/kcsrv
        cd kcsrv
        virtualenv -p /usr/bin/python3 .
    
    Now, enter the project's Virtual Environment. This contraption allows you to install packages locally, without conflicting with system packages. It's activated per-terminal, and must be activated before interacting with the project in any way:
    
        # Not a typo; `.` is an alias for `source`
        . bin/activate
    
    If you wish to exit it again, simply run:
    
        deactivate

1. **Install dependencies**
    
    Inside the `virtualenv`, simply run:
    
        pip install -r requirements.txt
    
    If you ever get issues with things not being found, try re-running this and seeing what happens.

1. **Set up the database**
    
    First, make sure you have a database user for the account you plan on running the server as. Postgres can automatically authorize as the user a process is running as, which means there's no need to put passwords in configuration files.
    
        sudo -u postgres createuser --superuser $USER
    
    Then, create a database for `kcsrv` to use:
    
        created kcsrv
    
    And fill it with data - repeat this after every upgrade:
    
        ./manage.py db upgrade   # Run migrations
        ./manage.py db update_db # Imports ship data into the database

    On your first run, you also need to create some base data:
    
        ./manage.py setup

1. **Download assets from DMM**

    This will take a while... you may want to re-run it after game patches, to stay up-to-date.
    
        ./manage.py dlassets

1.  **Create an account for yourself.**  
    Be sure to give yourself the `admin` and `staff` roles!
    Depending on your terminal, you may need to put the email address in quotation marks.
    
        ./manage.py user create <username> <email>
 
1.  **Run the development server.**  
    It will run on port 5000 by default.
    
        ./kcsrv.py